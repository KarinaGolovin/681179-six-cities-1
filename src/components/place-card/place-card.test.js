import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';
import history from '../../history';
import {Router} from 'react-router-dom';

it(`PlaceCard correctly renders`, () => {
  const component = renderer.create(
      <Router history={history}>
        <PlaceCard
          title={`Some text`}
          onClick={jest.fn()}
        />
      </Router>
  ).toJSON();

  expect(component).toMatchSnapshot();
});
