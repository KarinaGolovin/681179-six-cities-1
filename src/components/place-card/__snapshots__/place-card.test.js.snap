// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard correctly renders 1`] = `
<article
  className="place-card "
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__image-wrapper "
  >
    <a
      href="/offer/undefined"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        width={260}
      />
    </a>
  </div>
  <div
    className="place-card__info "
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="__bookmark-button  button"
        type="button"
      >
        <svg
          className="__bookmark-icon"
          height={19}
          width={18}
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="rating place-card__rating"
    >
      <div
        className="rating__stars place-card__stars"
      >
        <span
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/undefined"
        onClick={[Function]}
      >
        Some text
      </a>
    </h2>
    <p
      className="place-card__type"
    />
  </div>
</article>
`;
